qsimcirq/qsim_simulator.py

```diff
@@ -261,28 +261,36 @@ def simulate_sweep(
           qubit_order: Determines the canonical ordering of the qubits. This is
             often used in specifying the initial state, i.e. the ordering of the
             computational basis states.
-          initial_state: The initial state for the simulation. This is currently
-            unsupported in qsim; the recommended workaround is to prepend the
-            circuit with X gates as necessary.
+          initial_state: The initial state for the simulation. This can either
+            be an integer representing a pure state (e.g. 11010) or a numpy
+            array containing the full state vector. If none is provided, this
+            is assumed to be the all-zeros state.
 
       Returns:
           List of SimulationTrialResults for this run, one for each
           possible parameter resolver.
 
       Raises:
-          ValueError: if an initial_state is provided.
+          TypeError: if an invalid initial_state is provided.
       """
-    if initial_state is not None:
-      raise ValueError(
-        f'initial_state is not supported; received {initial_state}')
-
+    if initial_state is None:
+      initial_state = 0
+    if not isinstance(initial_state, (int, np.ndarray)):
+      raise TypeError('initial_state must be an int or state vector.')
     if not isinstance(program, qsimc.QSimCircuit):
       program = qsimc.QSimCircuit(program, device=program.device)
 
     options = {}
     options.update(self.qsim_options)
 
     param_resolvers = study.to_resolvers(params)
+    num_qubits = len(program.all_qubits())
+    input_vector = None
+    if isinstance(initial_state, int):
+      input_vector = np.zeros(2**num_qubits * 2, dtype=np.float32)
```

Comment:
Optional: the allocation of a temporary input vector can be avoided here. In this case, perhaps the order of relevant bits in `initial_state` should be reversed.

Reply:
Done. It appears that `SetAmpl` handles the necessary reversal.